### PR TITLE
Use Scalars[ID][input|output] as it works for better for client/server types

### PR DESCRIPTION
**Pull request by eddeee888**

### CODE REVIEW

File: src/schema/book/resolvers/Magazine.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import type { MagazineResolvers } from "./../../types.generated";
 export const Magazine: MagazineResolvers = {
   /* Implement Magazine resolver logic here */
+  id: ({ id }) => id, // Note 2: This is `string | number` 👍
```
Comment (n1ru4l) [inline src/schema/book/resolvers/Magazine.ts:4]: I think `id` here should be `string` as this is after the value coercion phase?

File: src/schema/types.generated.ts [hunk 2]
```diff
@@ -22,18 +22,30 @@ export type RequireFields<T, K extends keyof T> = Omit<T, K> & {
...
-  ID: string;
-  String: string;
-  Boolean: boolean;
-  Int: number;
-  Float: number;
-  DateTime: Date | string;
+  ID: {
+    input: string;
+    output: string | number;
+  };
```
Comment (n1ru4l) [inline src/schema/types.generated.ts]: For `ID` input should be `string | number` and output should be `string`

File: src/schema/book/resolvers/Magazine.ts (hunk 1 above)
Comment (eddeee888) [reply src/schema/book/resolvers/Magazine.ts:4]: I'm seeing this value being kept as-is from the resolver. I think the coercion happens just before the ID is returned to the client (i.e. just after this `id` function gets called). 

Here's how I tested it:

1. Return a `number` in `Query.readable`
2. Log out the type in `Magazine.id` resolver. I'm seeing it's still a `number`

Here's the code:

Screenshot 2023-05-02 at 8 56 15 pm

---

Here's the query that I use:

```graphql
query Test($id1:ID!) {
  mag: readable(id: $id1) {
    ... on Magazine {
      id
    }
    ... on  ShortNovel{
      id
    }
  }
}
```

Let me know if you are seeing something different 🙂

File: src/schema/types.generated.ts (hunk 2 above)
Comment (eddeee888) [reply src/schema/types.generated.ts]: Yes, I think that's right if we are generating `ID` type for the client.

For the server, I'm seeing that when an ID input gets into the resolver, it's a `string`.  I think this the coercion happens just before ID gets to the resolver.

This matches the proposed `Scalars` types and related Query arg types:
```ts
export type QueryBookArgs = {
  id: Scalars["ID"]["input"]; // resolvers reference this, which is a `string`
};
```

**Pull request closed without merging**